stock-quote/src/main/java/com/whz/stockquote/StockQuotePublisher.java

```diff
@@ -46,26 +96,51 @@ public void publish(List<String> symbols) throws IOException {
         channel.basicPublish(
             exchangeName, "", null, answerJsonString.getBytes(StandardCharsets.UTF_8));
       }
-    } catch (TimeoutException e) {
-      LOGGER.severe(e.toString());
+    } catch (IOException e) {
+      logger.error(e.toString());
     }
   }
 
-  public void run(List<String> symbols) throws IOException, InterruptedException {
-    String randomPublishIntervalString = System.getenv("SET_RANDOM_PUBLISH_INTERVAL");
+  /**
+   * Creates a connection and a channel to publish messages to RabbitMQ. Publish messages to
+   * RabbitMQ continuously. The publishing interval may be a random number within zero and the
+   * {@code publishInterval} as an upper limit.
+   *
+   * @param symbols list of official stock symbols
+   * @param publishInterval time interval in seconds
+   * @param publishRandomly true if publishing interval should be random, false otherwise
+   * @since 1.0.0
+   */
+  public void run(List<String> symbols, int publishInterval, boolean publishRandomly) {
+    try {
+      connection = connectionFactory.newConnection();
+      Channel channel = connection.createChannel();
+
+      int actualPublishInterval = publishInterval * 1000;
 
-    if (Boolean.parseBoolean(randomPublishIntervalString)) {
-      while (true) {
-        publish(symbols);
-        Thread.sleep((long) (Math.random() * 3000));
-      }
-    } else {
-      LOGGER.warning(
-          "Environment variable 'SET_RANDOM_PUBLISH_INTERVAL' either incorrect or not set!");
       while (true) {
-        publish(symbols);
-        Thread.sleep(30000);
+        publish(symbols, channel);
+
+        if (publishRandomly) {
+          actualPublishInterval = (int) (Math.random() * publishInterval * 1000);
+        }
+        logger.info(String.format("Publishing interval: %dms", actualPublishInterval));
+
+        try {
+          Thread.sleep(actualPublishInterval);
+        } catch (InterruptedException e) {
+          logger.error(e.toString());
+        }
       }
+    } catch (IOException | TimeoutException e) {
+      logger.error(e.toString());
```

Comment:
To get the whole stacktrace:
```suggestion
      logger.error("your message", e);
```

You use it in many places in the code, please correct those too.